modules/frontend/metrics_query_range_sharder.go

```diff
@@ -26,11 +26,12 @@ import (
 )
 
 type queryRangeSharder struct {
-	next      pipeline.AsyncRoundTripper[combiner.PipelineResponse]
-	reader    tempodb.Reader
-	overrides overrides.Interface
-	cfg       QueryRangeSharderConfig
-	logger    log.Logger
+	next        pipeline.AsyncRoundTripper[combiner.PipelineResponse]
+	reader      tempodb.Reader
+	overrides   overrides.Interface
+	cfg         QueryRangeSharderConfig
+	logger      log.Logger
+	instantMode bool
```

Comment:
What do you think about not splitting the pipelines and using [traceql.IsInstant(req)](https://github.com/grafana/tempo/blob/main/pkg/traceql/engine_metrics.go#L135) ?

Reply:
We then need to compile the query, I think it's not that bad to split the pipeline in case we want to change something in the future. I don't have a strong opinion about that, both options are ok